### DISCUSSION TITLE
Authentication and User Management

### DISCUSSION THREAD

**Discussion by cuanjooste** [Azure/static-web-apps#747] [Q&A] [answered]
Hi

I am unsure if I am misunderstanding something or if I just missed the relevant information, but I am having some issues wrapping my head around authentication.

1) Is there a webhook or something to trigger the first time a user authenticates. For example calling a function to create a new user in a database.
2) Is there anywhere in azure portal where one can manage or see users who has authenticated?

Thanks in advance

**Comment by anthonychu** [comment, marked as answer]
You can run a function each time a user successfully authenticates: https://docs.microsoft.com/en-us/azure/static-web-apps/authentication-authorization?tabs=function#role-management

The function is mainly for looking up custom roles, but you don’t need to return any roles. It’ll be passed details about the user that you can save to your database. There’s nothing built-in to see this information.

**Reply by cuanjooste** [reply]
Thank you for the response, I will give it a try.
